### PR TITLE
add EngineCpuUtilizationAverage metric

### PR DESCRIPTION
**Pull request by austincunningham**
Co-authored-by: darahayes

## Overview

Jira: https://issues.redhat.com/browse/INTLY-7234

## Verification

- Clone this branch
- Run `make cluster/prepare`
- Run `make run`
- Run `make cluster/seed/managed/redis`
- Ensure Once Redis is deployed on aws check the metrics for this metric

```
curl localhost:8383/metrics | grep cro_redis_engine_cpu_utilization_average
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
# HELP cro_redis_engine_cpu_utilization_average The percentage of CPU utilization. Units: Percent
# TYPE cro_redis_engine_cpu_utilization_average gauge
cro_redis_engine_cpu_utilization_average{clusterID="mw-collab-byoc-z5xz6",instanceID="mwcollabbyocz5xz6redhatrhmioperator-o2wi-001",namespace="redhat-rhmi-operator",productName="productName",resourceID="example-redis",strategy="aws-elasticache"} 9.25
cro_redis_engine_cpu_utilization_average{clusterID="mw-collab-byoc-z5xz6",instanceID="mwcollabbyocz5xz6redhatrhmioperator-o2wi-002",namespace="redhat-rhmi-operator",productName="productName",resourceID="example-redis",strategy="aws-elasticache"} 1
100 74160    0 74160    0     0  27.8M      0 --:--:-- --:--:-- --:--:-- 35.3M

## Checklist
- [ ] This PR includes a change to an instance type

### CODE REVIEW

File: pkg/controller/cloudmetrics/cloudmetrics_controller.go [hunk 1]
```diff
@@ -32,6 +32,7 @@ const (
 	redisMemoryUsagePercentageAverage = "cro_redis_memory_usage_percentage_average"
 	redisFreeableMemoryAverage        = "cro_redis_freeable_memory_average"
 	redisCPUUtilizationAverage        = "cro_redis_cpu_utilization_average"
+	redisEngineCPUUtilizationAverage        = "cro_redis_engine_cpu_utilization_average"
```
Comment (darahayes) [inline pkg/controller/cloudmetrics/cloudmetrics_controller.go:35]: Perhaps the alignment needs to be formatted here. At least in my editor this happens automatically.

Comment (laurafitzgerald) [reply pkg/controller/cloudmetrics/cloudmetrics_controller.go:35]: make code/fix would resolve it too. 🥳

**Comment by laurafitzgerald** [comment]
Verification: Followed the steps in the description.

Verify that there was one alert with the name `cro_redis_engine_cpu_utilization_average`

Screenshot 2020-08-12 at 16 38 50

I noticed that there are two entries in the desription. Should I expect one or two entires (one per node) or did you guys just have an additional redis running?

The change itself looks good. Could the commits be squashed?

Also to do co-autoring at a commit level if you add the following to the commit message it references the user on the commit also. :D 

Co-authored-by: githubusername <email@redhat.com>

**Comment by austincunningham** [comment]
@laurafitzgerald would expect two , one per node. the example was from cro running in rhmi.

**Comment by austincunningham** [comment]
closing due to messed up rebase.

**Comment by laurafitzgerald** [comment]
> @laurafitzgerald would expect two , one per node. the example was from cro running in rhmi.

Cool. I ran it from cro directly. When I check it now there's two. Happy to run it again when there's a new pr. 👍

**Comment by austincunningham** [comment]
rebase went badly cherry-picked to https://github.com/integr8ly/cloud-resource-operator/pull/176

**Pull request closed without merging**